modules/loader/src/main/scala/com/snowplowanalytics/snowplow/postgres/config/ConfigPreCheck.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2020-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.postgres.config
+
+import cats.effect.Sync
+import cats.data.EitherT
+import cats.implicits._
+
+import java.nio.file.Files
+import com.snowplowanalytics.snowplow.postgres.config.LoaderConfig.Source.LocalFS
+
+import scala.util.Try
+
+/**
+  * Check if given configuration meets necessary requirements
+  */
+object ConfigPreCheck {
+  def apply[F[_]: Sync](cli: Cli[F]): EitherT[F, String, Cli[F]] =
+    EitherT(Sync[F].delay {
+      cli.config.input match {
+        // If configuration is for local source, check if given path exists
+        case LocalFS(pathInfo) =>
+          Either.fromTry(Try {
```

Comment:
Same thing about `Either.catchNonFatal`. But ideally I'd even split this all into smaller referentially-transparent actions, such as:

```scala
def fileExists(pathInfo: PathInfo): F[Boolean] = ???

cli.config.input match {
  case LocalFS(pathInfo) =>
    fileExists(pathInfo)
      .attemptT
      .leftMap(_.toString)
      .ensure(s"Local source path [${pathInfo.allPath}] does not exist")(identity)
      .as(cli)
```

Another thing is that I think it can live right in `Cli` module?